vaadin-ordered-layout-flow-parent/vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/ThemableLayout.java

```diff
@@ -93,16 +94,58 @@ default boolean isPadding() {
      *            it if {@code false}
      */
     default void setSpacing(boolean spacing) {
+        if (!spacing) {
+            getElement().getStyle().remove("gap");
+        }
         getThemeList().set("spacing", spacing);
     }
 
     /**
-     * Shows if {@code spacing} theme setting is applied to the component.
+     * Shows if {@code spacing} setting is applied to the component, either by
+     * setting the {@code spacing} theme or by setting the {@code gap} style.
      *
      * @return {@code true} if theme setting is applied, {@code false} otherwise
      */
     default boolean isSpacing() {
-        return getThemeList().contains("spacing");
+        return getThemeList().contains("spacing")
+                || getElement().getStyle().has("gap");
+    }
+
+    /**
+     * Sets the spacing between the components inside the layout.
+     *
+     * @param spacing
+     *            the spacing between the components. The value must be a valid
+     *            CSS length, i.e. must provide a unit (e.g. "1px", "1rem",
+     *            "1%") for values other than 0.
+     */
+    default void setSpacing(String spacing) {
+        setSpacing(true);
+        getElement().getStyle().set("--_gap", spacing);
```

Comment:
We set `--_gap` here but use `gap` elsewhere, should it be aligned? Also, unit tests seem to fail.

Reply:
Good catch. This is a leftover of the implementation that was supposed to use this `--_gap` variable on the client side. Changed to `gap` instead.